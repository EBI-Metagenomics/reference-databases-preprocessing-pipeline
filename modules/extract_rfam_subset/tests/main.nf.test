nextflow_process {

    name "Test Process EXTRACT_RFAM_SUBSET"
    script "modules/extract_rfam_subset/main.nf"
    process "EXTRACT_RFAM_SUBSET"
    tag "modules"
    tag "extract_rfam_subset"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = file('${projectDir}/modules/extract_rfam_subset/tests/data/test.cm', checkIfExists: true)
                input[1] = file('${projectDir}/modules/extract_rfam_subset/tests/data/temp_ribo.claninfo', checkIfExists: true)
                input[2] = params.rfam_version
                input[3] = params.rfam_label
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.ribo_cm).match("ribo_cm") },
                { assert snapshot(process.out.ribo_claninfo).match("ribo_claninfo") }
            )
        }

    }

}
