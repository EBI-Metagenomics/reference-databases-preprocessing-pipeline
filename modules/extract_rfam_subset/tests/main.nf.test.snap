{
    "ribo_claninfo": {
        "content": [
            [
                "ribo.clan_info:md5,ed55b0130da268e5deca604fa34da618"
            ]
        ],
        "timestamp": "2023-12-12T16:44:40.410580931"
    },
    "ribo_cm": {
        "content": [
            [
                "ribo.cm:md5,fafc048b3c74460583726cc91959dec7"
            ]
        ],
        "timestamp": "2023-12-13T08:42:07.138270818"
    }
}